python/idsse_common/idsse/common/http_utils.py

```diff
@@ -0,0 +1,216 @@
+
+import requests
+
+from .path_builder import PathBuilder
+from .utils import TimeDelta, datetime_gen
+
+logger = logging.getLogger(__name__)
+
+# pylint: disable=duplicate-code, broad-exception-caught
+
+class HttpUtils:
+    """http Utility Class - Used by DAS for file downloads"""
+
+    def __init__(self,
+                 basedir: str,
+                 subdir: str,
+                 file_base: str,
+                 file_ext: str) -> None:
+        self.path_builder = PathBuilder(basedir, subdir, file_base, file_ext)
+
+    def get_path(self, issue: datetime, valid: datetime) -> str:
+        """Delegates to instant PathBuilder to get full path given issue and valid
+
+        Args:
+            issue (datetime): Issue date time
+            valid (datetime): Valid date time
+
+        Returns:
+            str: Absolute path to file or object
+        """
+        lead = TimeDelta(valid-issue)
+        return self.path_builder.build_path(issue=issue, valid=valid, lead=lead)
+
+    @staticmethod
+    def http_ls(url: str, prepend_path: bool = True) -> Sequence[str]:
+        """Execute an 'ls' on the http(s) server
+        Args:
+            url (str): URL
+            prepend_path (bool): Add URL+ to the filename
+        Returns:
+            Sequence[str]: The results from executing a request get on passed url
+        """
+        try:
+            files = []
+            response = requests.get(url, timeout=5)
+            response.raise_for_status()  # Raise an exception for bad status codes
+
+            for line in response.text.splitlines():
+                if 'href="' in line:
+                    filename = line.split('href="')[1].split('"')[0]
+                    if not filename.endswith('/'):  # Exclude directories
+                        files.append(filename)
+
+        except requests.exceptions.RequestException as exp:
+            logger.warning('Unable to query supplied URL: %s', str(exp))
+            return []
+        if prepend_path:
+            return [os.path.join(url, filename) for filename in files]
+        return files
+
+    @staticmethod
+    def http_cp(url: str, dest: str) -> bool:
+        """Execute http request download from URL to dest.
+
+        Args:
+            url (str): URL to the object to be copied
+            dest (str): The destination location
+        Returns:
+            bool: Returns True if copy is successful
+        """
+        try:
+            with requests.get(os.path.join(url), timeout=5, stream=True) as response:
+                # Check if the request was successful
+                if response.status_code == 200:
+                    # Open a file in binary write mode
+                    with open(dest, "wb") as file:
+                        shutil.copyfileobj(response.raw, file)
+                    return True
+
+                logger.debug('copy fail: request status code: %s', response.status_code)
+                return False
+        except Exception:
```

Comment:
```suggestion
        except Exception:  # pylint: disable=broad-exception-caught
```
But maybe we can figure out what type of exceptions can actually be raise and except those specifically